### PR TITLE
[APB-3509]: Add search not found error and add internal HMRC styling

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/uk/gov/hmrc/agentstatuschangefrontend/connectors/AgentServicesAccountConnector.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+@Singleton
+class AgentServicesAccountConnector @Inject()(appConfig: AppConfig, http: HttpPost with HttpGet, metrics: Metrics)
+    extends HttpAPIMonitor {
+
+  override val kenshooRegistry: MetricRegistry = metrics.defaultRegistry
+
+  def getAgencyNameByArn(arn: Arn)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[String]] =
+    monitor(s"ConsumedAPI-Get-AgencyName-GET") {
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentstatuschangefrontend/connectors/AgentServicesAccountConnector.scala]: Code for `getAgencyNameByArn` is identical to `getAgencyNameByUtr` except for the `/arn/` or `/utr/` bit in the url. Potential to have two thin methods delegate with different parameters to just one method with the implementation.

Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentstatuschangefrontend/connectors/AgentServicesAccountConnector.scala]: 👍

File: app/uk/gov/hmrc/agentstatuschangefrontend/journeys/AgentStatusChangeJourneyModel.scala [hunk 2]
```diff
@@ -32,45 +32,69 @@ object AgentStatusChangeJourneyModel extends JourneyModel {
...
     import uk.gov.hmrc.agentstatuschangefrontend.journeys.AgentStatusChangeJourneyModel.State._
 
-    type GetAgentDetails = String => Future[Option[AgentDetails]]
+    type GetStatus = String => Future[Option[AgentStatus]]
+    type GetAgentNameArn = Arn => Future[Option[String]]
+    type GetAgentNameUtr = Utr => Future[Option[String]]
 
-    def submitSearch(getAgentDetails: GetAgentDetails)(strideId: String)(providedAgentIdentifier: String) = Transition {
+    def submitSearch(getStatus: GetStatus)(getAgentNameArn: GetAgentNameArn)(getAgentNameUtr: GetAgentNameUtr)(
+      strideId: String)(providedAgentIdentifier: String) = Transition {
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentstatuschangefrontend/journeys/AgentStatusChangeJourneyModel.scala]: Potential for code reuse in here. Same pattern is repeated 4 times.

Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentstatuschangefrontend/journeys/AgentStatusChangeJourneyModel.scala]: yes you're right, will fix

File: it/uk/gov/hmrc/agentstatuschangefrontend/controllers/AgentStatusChangeControllerISpec.scala [hunk 3]
```diff
@@ -37,28 +38,62 @@ class AgentStatusChangeControllerISpec extends BaseISpec {
...
+        givenAgencyNameUtr(Utr("3110118001"), "Mr Agency Name")
+        givenAgentStatus("3110118001", "suspended")
+        givenAuthorisedFor("maintain agent relationships", "PrivilegedApplication", "StrideUserId")
+        journeyState.set(Search, Nil)
+        val result = controller.submitSearch(FakeRequest().withFormUrlEncodedBody("agentIdentifier" -> "3110118001"))
+        status(result) shouldBe 303
+        redirectLocation(result) shouldBe Some(routes.AgentStatusChangeController.showDetails().url)
+      }
+
+      "redirect to search page if agent details not found for Arn" in {
```
Comment (johnrevill) [inline it/uk/gov/hmrc/agentstatuschangefrontend/controllers/AgentStatusChangeControllerISpec.scala]: Pedantic: "redirect to search error page"? Same below on line 82

**Pull request merged**